engine/src/main/java/org/pentaho/di/imp/Import.java

```diff
@@ -0,0 +1,479 @@
+import org.pentaho.di.repository.RepositoryImportFeedbackInterface;
+import org.pentaho.di.repository.RepositoryMeta;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.version.BuildVersion;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+public class Import {
+  private static final Class<?> PKG = Import.class; // i18n
+
+  public static final String STRING_IMPORT = "Import";
+  public static final String ROOT_DIRECTORY = "/";
+  public static final String PUBLIC_DIRECTORY = "public";
+  public static final String HOME_DIRECTORY = "home";
+
+  private static class ImportFeedback implements RepositoryImportFeedbackInterface, HasOverwritePrompter {
+    private final LogChannelInterface log;
+    private final boolean continueOnError;
+    private final boolean replace;
+    private final BufferedReader reader;
+
+    public ImportFeedback( LogChannelInterface log, boolean continueOnError, boolean replace, BufferedReader reader ) {
+      this.log = log;
+      this.continueOnError = continueOnError;
+      this.replace = replace;
+      this.reader = reader;
+    }
+
+    @Override
+    public void updateDisplay() {
+    }
+
+    @Override
+    public boolean transOverwritePrompt( TransMeta transMeta ) {
+      return replace;
+    }
+
+    @Override
+    public void showError( String title, String message, Exception e ) {
+      log.logError( title + " : " + message, e );
+    }
+
+    @Override
+    public void setLabel( String labelText ) {
+      log.logBasic( labelText );
+    }
+
+    @Override
+    public boolean jobOverwritePrompt( JobMeta jobMeta ) {
+      return replace;
+    }
+
+    @Override
+    public boolean askContinueOnErrorQuestion( String title, String message ) {
+      return continueOnError;
+    }
+
+    @Override
+    public void addLog( String line ) {
+      log.logBasic( line );
+    }
+
+    @Override
+    public boolean isAskingOverwriteConfirmation() {
+      return false;
+    }
+
+    @Override
+    public OverwritePrompter getOverwritePrompter() {
+      return new OverwritePrompter() {
+        private final String yes = BaseMessages.getString( PKG, "Import.Yes" );
+        private final String no = BaseMessages.getString( PKG, "Import.No" );
+        private final String none = BaseMessages.getString( PKG, "Import.None" );
+        private final String all = BaseMessages.getString( PKG, "Import.All" );
+        private final String prompt = "[" + yes + "," + no + "," + none + "," + all + "]";
+
+        @Override
+        public boolean overwritePrompt( String message, String rememberText, String rememberPropertyName ) {
+          log.logBasic( message );
+          String line;
+          Boolean result = null;
+          boolean remember = false;
+          while ( result == null ) {
+            log.logBasic( prompt );
+            try {
+              line = reader.readLine().trim();
+            } catch ( IOException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)